1_fetch/src/get_daily_nwis_data.R

```diff
@@ -0,0 +1,63 @@
+  #' @examples 
+  #' get_daily_nwis_data(site_info = daily_sites,parameter="00300",stat_cd_select="00003")
+  
+  message(sprintf('Retrieving daily data for %s', site_info$site_no))
+
+  # Download daily data
+  site_data <- dataRetrieval::readNWISdv(
+    siteNumbers = site_info$site_no,parameterCd=parameter,statCd=stat_cd_select,startDate = "",endDate = "") %>%
+    dataRetrieval::renameNWISColumns(p00300="Value",p00095="Value")
+  
+  # Munge column names for some sites with different or unusually-named value columns
+  if(length(grep("Value_cd",names(site_data)))>1){
+    site_data <- switch(site_info$site_no[1],
+                        "01463500" = site_data %>%
+                          mutate(Value_merged = coalesce(`Value`,`..2.._Value`),
+                                 Value_cd_merged = coalesce(`Value_cd`,`..2.._Value_cd`),
+                                 Value_Max_merged = coalesce(`Value_Max`,`..2.._Value_Max`),
+                                 Value_Max_cd_merged = coalesce(`Value_Max_cd`,`..2.._Value_Max_cd`)) %>%
+                          select(agency_cd,site_no,Date,Value_merged,Value_cd_merged,Value_Max_merged,Value_Max_cd_merged) %>%
+                          rename("Value"="Value_merged","Value_cd" = "Value_cd_merged","Value_Max"="Value_Max_merged","Value_Max_cd"="Value_Max_cd_merged"),
+                        "01467200" = site_data %>%
+                          mutate(Value_merged = coalesce(`Value`,`ISM.Test.Bed...ISM.barge._Value`),
+                                 Value_cd_merged = coalesce(`Value_cd`,`ISM.Test.Bed...ISM.barge._Value_cd`)) %>%
+                          select(agency_cd,site_no,Date,Value_merged,Value_cd_merged,Value_Max,Value_Max_cd) %>%
+                          rename("Value"="Value_merged","Value_cd"="Value_cd_merged"),
+                        "01412350" = site_data %>%
+                          mutate(Value = `.Discontinued._Value`,
+                                 Value_cd = `.Discontinued._Value_cd`,
+                                 Value_Max = `.Discontinued._Value_Max`,
+                                 Value_Max_cd = `.Discontinued._Value_Max_cd`) %>%
+                          select(agency_cd,site_no,Date,Value,Value_cd,Value_Max,Value_Max_cd))
+  }
+  
+  # If no max value reported, create empty column:
+  if(!('Value_Max' %in% names(site_data))){
+    site_data <- site_data %>%
+      add_column(Value_Max = NA,
+                 Value_Max_cd = NA)
+  }
+  
+  # Filter daily data
+  site_data_out <- site_data %>%
+           # omit rows with no data
+    filter(!(is.na(Value) & is.na(Value_Max)),
+           # omit rows where daily mean > daily max
+           (is.na(Value)|is.na(Value_Max)|(!Value > Value_Max)),
+           # omit rows with undesired data quality codes
+           !(Value_cd %in% c("P Eqp","P Mnt")),
+           !(Value_Max_cd %in% c("P Eqp","P Mnt"))) %>%
```

Comment:
I have seen records that omit spaces, use lowercase, etc. for these codes (I haven't checked if that's the case for our project sites). Maybe grepping for "eqp" and "mnt" is better to catch these oddities (with `ignore.case = TRUE`).

Reply:
I noticed a filter for these in other scripts above, too, so comment also applies wherever they appear. Do you know a way to search the whole repo for text in R? Looks like it is possible within GitHub

Reply:
Great idea, I'll reformat this to use `grep()` instead. I haven't seen those other variations yet, but I definitely believe you that they're out there! The `grep()` approach will be more robust in any case.

Reply:
You can search in Rstudio, at least: Ctrl+Shift+F opens a "Find in Files" window that you can use to search for text across files in an R project.